client/landing/domains/registrant-verification/index.jsx

```diff
@@ -0,0 +1,249 @@
+		};
+	};
+
+	getEmailMismatchState = () => {
+		const { domain, email, translate } = this.props;
+
+		return {
+			message: translate(
+				'{{strong}}%(email)s{{/strong}} is different from the email we have on record for {{strong}}%(domain)s{{/strong}}.',
+				{
+					args: {
+						email: email,
+						domain: domain,
+					},
+					components: {
+						strong: <strong />,
+					},
+				}
+			),
+		};
+	};
+
+	getKeySystemsErrorState = errorMessage => {
+		if (
+			'Invalid attribute value; Contact verification already confirmed, nothing to do' ===
+			errorMessage
+		) {
+			const { domain, email, translate } = this.props;
+
+			return {
+				illustration: '/calypso/images/illustrations/illustration-ok.svg',
+				title: translate( 'Already verified.' ),
+				message: translate(
+					"You've already verified {{strong}}%(email)s{{/strong}} for:{{br /}}{{strong}}%(domain)s{{/strong}}.",
+					{
+						args: {
+							email: email,
+							domain: domain,
+						},
+						components: {
+							strong: <strong />,
+							br: <br />,
+						},
+					}
+				),
+				footer: translate( 'All done. You can close this window now.' ),
+			};
+		}
+	};
+
+	getResendEmailErrorState = () => {
+		const { translate } = this.props;
+
+		return {
+			message: translate( "Sorry, we weren't able to re-send the email." ),
+		};
+	};
+
+	getDefaultErrorState = () => {
+		const { translate } = this.props;
+		const defaultErrorFooter = translate(
+			"If you're having trouble verifying your contact information, please {{a}}{{strong}}contact support{{/strong}}{{/a}}.",
+			{
+				components: {
+					a: <a href={ CALYPSO_CONTACT } />,
+					strong: <strong />,
+				},
+			}
+		);
+
+		return {
+			title: translate( 'Uh oh!' ),
+			illustration: '/calypso/images/illustrations/error.svg',
+			message: translate( 'Hmm. Something went wrong.' ),
+			actionTitle: null,
+			actionCallback: null,
+			footer: defaultErrorFooter,
+			isLoading: false,
+		};
+	};
+
+	setErrorState = error => {
+		let errorState;
+
+		switch ( error.error ) {
+			case 'email_mismatch':
+				errorState = this.getEmailMismatchState();
+				break;
+
+			case 'unauthorized':
+				errorState = this.getExpiredState();
+				break;
+
+			case 'Key_Systems_error':
+				errorState = this.getKeySystemsErrorState( error.message );
+				break;
+
+			case 'resend_email_failed':
+				errorState = this.getResendEmailErrorState();
+				break;
+		}
+
+		this.setState( {
+			...this.getDefaultErrorState(),
+			...errorState,
+		} );
+	};
+
+	resendEmail = () => {
+		const { domain, translate } = this.props;
+
+		this.setState( this.getLoadingState() );
+
+		wpcom.resendIcannVerification( domain, error => {
+			if ( error ) {
+				this.setErrorState( { error: 'resend_email_failed' } );
+			} else {
+				this.setState( {
+					illustration: '/calypso/images/illustrations/illustration-ok.svg',
+					title: translate( 'Email sent!' ),
+					message: translate( 'Check your email.' ),
```

Comment:
How about we show the email we've sent the email to? We do that in Domain Management and it makes sense here too. File it under "nice to have things" as it needs changes on the backend to return that info (email) from the backend on success.
Not super high-prio, since this whole flow should be an edge case.